yaksh/models.py

```diff
@@ -1216,11 +1216,18 @@ def can_attempt_now(self, user, course_id):
             )
             if last_attempt:
                 time_lag = (timezone.now() - last_attempt.start_time).total_seconds() / 3600
-                return time_lag >= self.quiz.time_between_attempts
+                can_attempt = time_lag >= self.quiz.time_between_attempts
```

Comment:
Instead this thing can be checked in views. This way this code remains intact.

Reply:
I just changed the return values, earlier it used to be just a True/False now the error message also gets set here.